api/src/infrastructure/docker.rs

```diff
@@ -164,35 +175,43 @@ impl DockerInfrastructure {
         debug!("Creating network for app {}.", app_name);
 
         let network_create_info = docker
-            .networks()
-            .create(&NetworkCreateOptions::builder(network_name.as_ref()).build())
+            .create_network(CreateNetworkOptions::<&str> {
+                name: &network_name,
+                ..Default::default()
+            })
             .await?;
+        let network_id = network_create_info
+            .id
+            .expect("id is mandatory for a Docker Network.");
 
-        debug!(
-            "Created network for app {} with id {}",
-            app_name, network_create_info.id
-        );
+        debug!("Created network for app {} with id {network_id}", app_name);
 
-        Ok(network_create_info.id)
+        Ok(network_id)
     }
 
-    async fn connect_traefik(&self, network_id: &String) -> Result<(), ShipLiftError> {
-        let docker = Docker::new();
+    async fn connect_traefik(&self, network_id: &String) -> Result<(), BollardError> {
+        let docker = Docker::connect_with_socket_defaults()?;
 
         let containers = docker
-            .containers()
-            .list(&ContainerListOptions::builder().build())
+            .list_containers(Some(ListContainersOptions::<&str> {
+                all: true,
+                filters: HashMap::from([("label", vec!["org.opencontainers.image.title=Traefik"])]),
```

Comment:
That is new. Are we sure that Traefik 1.7 has this label? (unfortunately, Docker depends still on 1.7, see #23)

Reply:
I will do some further digging around and confirm from the official image if these labels are always present across versions. My traefik 1.7 local container did contain this label, hence the approach.

Reply:
Using image title is a hardcode, there are chances it will change with version.
Another option is to use filter `ancestor` with image name. However this does not support partial match i.e `traefik:v1.7-alpine` is the only valid match. So again version dependency.
A manual find is the best approach, i will revert to original.